### PR TITLE
[DEVTOOLING-854] - Added name and type to flow Export

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: genesyscloud/architect_flow/resource_genesyscloud_architect_flow_schema.go [hunk 1]
```diff
@@ -49,6 +49,18 @@ func ResourceArchitectFlow() *schema.Resource {
 		},
 		SchemaVersion: 1,
 		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Flow name.",
+				Type:        schema.TypeString,
+				Optional:    false,
```
Comment (HemanthDogiparthi12) [inline genesyscloud/architect_flow/resource_genesyscloud_architect_flow_schema.go:55]: optional to true

File: docs/resources/flow.md [hunk 2]
```diff
@@ -72,4 +72,6 @@ resource "genesyscloud_flow" "flow" {
 ### Read-Only
 
 - `id` (String) The ID of this resource.
+- `name` (String) Flow name.
+- `type` (String) Flow type. Valid options: bot, commonmodule, digitalbot, inboundcall, inboundchat, inboundemail, inboundshortmessage, outboundcall, inqueuecall, inqueueemail, inqueueshortmessage, speech, securecall, surveyinvite, voice, voicemail, voicesurvey, workflow, workitem
```
Comment (HemanthDogiparthi12) [inline docs/resources/flow.md:76]: there is another comment from Adam. https://inindca.atlassian.net/browse/DEVTOOLING-854?focusedCommentId=5790355 to extend this to datasource too. but looks like datasource already have it in its schema. just reverify if you are able to export datasource with filename and type intact. ( use replacewithdatasource option for export)

File: genesyscloud/architect_flow/resource_genesyscloud_architect_flow_schema.go [hunk 3]
```diff
@@ -49,6 +49,18 @@ func ResourceArchitectFlow() *schema.Resource {
...
+				Computed:    true,
+			},
+			"type": {
+				Description: "Flow type. Valid options: " + strings.Join(validFlowTypes, ", "),
```
Comment (HemanthDogiparthi12) [inline genesyscloud/architect_flow/resource_genesyscloud_architect_flow_schema.go:59]: though its optional, you can still add validatorfunc similar to datasource schema of genesyscloud_flow.

File: docs/resources/flow.md (hunk 2 above)
Comment (BrianMoyles) [reply docs/resources/flow.md:76]: Verified

**Review by HemanthDogiparthi12 — APPROVED** [review]

**Review by carnellj-genesys — APPROVED** [review]

**Pull request merged**